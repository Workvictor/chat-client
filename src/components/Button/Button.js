import React from 'react';
import styled from 'styled-components';


const Wrapper = styled.button`
  padding: 0.25rem 0.5rem;
  border: none;
  background: #fff;
  cursor: pointer;
  border-radius: 0.25rem;
  &.primary{
    background-color: Plum;
  }
  &.warning{
    background-color: Tomato;
    border: 1px solid OrangeRed
  }
`;

export const presetList = [
  `primary`,
  `warning`
];

const getPreset = props =>
  Object.keys(props)
    .filter(prop => presetList.includes(prop))
    .reduce((result, prop) =>`${result} ${prop}`, ``);

export const Button = props => {
  const { children } = props;
  const preset = getPreset(props);
  const className = (props.className || ``).concat(preset);
  return (
    <Wrapper
      {...{ className }}
      onClick={props.onClick}
      onMouseDown={props.onMouseDown}
      onMouseUp={props.onMouseUp}
    >
      {children}
    </Wrapper>
  )
}