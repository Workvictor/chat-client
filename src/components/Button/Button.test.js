import React from 'react';
import { shallow } from 'enzyme';
import { Button, presetList } from './Button';
import renderer from 'react-test-renderer';


describe(`Компонент Button`, () => {
  it('должен рендериться без ошибок', function () {
    shallow(<Button>test</Button>);
  });

  test('соответствовать снимку', () => {
    const tree = shallow(<Button>test</Button>);
    expect(tree).toMatchSnapshot();
  });

  presetList.forEach(presetName=>{
    it(`должен поддерживать пресет ${presetName}` , () => {
      const preset = shallow(<Button {...{[presetName]:true}}>test</Button>);
      expect(preset.is(`.${presetName}`)).toBe(true);
    });
  })

  it(`должен передавать className` , () => {
    const test = shallow(<Button className={`test`}>test</Button>);
    expect(test.is(`.test`)).toBe(true);
  });

});