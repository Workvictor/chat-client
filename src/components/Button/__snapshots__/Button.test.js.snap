// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button соответствовать снимку 1`] = `
<styled.button
  className=""
>
  test
</styled.button>
`;
